dev/ci/scripts/utils/rocotostat.py

```diff
@@ -88,18 +210,24 @@ def input_args():
 
 def rocotostat_summary(rocotostat):
     """
-    rocoto_summary Run rocotostat and process its output.
+    Execute rocotostat with summary flag and analyze workflow cycle completion status.
 
-    rocoto_summary(rocotostat) adds a default argument '--summary' to the rocotostat
-    command, runs it, and processes its output to return a dictionary with the total
-    number of cycles and the number of cycles marked as 'Done'.
+    This function invokes the rocotostat command with the '--summary' argument to retrieve high-level workflow information,
+    then processes the output to extract cycle completion statistics. It parses the summary output to determine the total
+    number of workflow cycles and counts how many cycles have reached the 'Done' state, providing essential metrics for
+    workflow progress monitoring and completion assessment.
 
-    Input:
-    rocotostat - The rocotostat command.
+    Parameters
+    ----------
+    rocotostat : callable
+        The rocotostat command object configured with workflow and database file paths for execution.
 
-    Output:
-    rocoto_status - A dictionary with the total number of cycles and the number of cycles marked as 'Done'.
+    Returns
+    -------
+    rocoto_status : dict
+        Dictionary containing 'CYCLES_TOTAL' (total cycles) and 'CYCLES_DONE' (completed cycles) for status tracking.
     """
+
     rocotostat = copy.deepcopy(rocotostat)
     rocotostat.add_default_arg('--summary')
     rocotostat_output = attempt_multiple_times(lambda: rocotostat(output=str), 3, 120, ProcessError)
```

Comment:
Instead of creating a deepcopy here, you can just pass rocotostat with the `--summary` argument:

```suggestion
    rocotostat_output = attempt_multiple_times(lambda: rocotostat(["--summary"], output=str), 3, 120, ProcessError)

Reply:
I recall going down that path and it hasn't worked in the past.  I'll revisit this and see what the issue is.

Reply:
Yes that suggestion worked